hex.pm release

mremond opened the pull request:
Hello Alexey,

I prepared a release of sqlite3 for hex.pm (as version 1.1.5) and here is what I used to prepare the package. We needed that to release ejabberd on hex.pm as all dependencies must be on hex.pm as well.

I did submit it with following command, using rebar3 only for submission to hex.pm:

```
rebar3 hex publish
```

Please, let me know if you want to manage the project on hex.pm and I will transfer you the right as well.

Cheers,

mremond committed:
Metadata for Hex.pm

mremond committed:
Metadata ready for hex.pm release

alexeyr commented on src/sqlite3.app.src:
```diff
@@ -5,5 +5,10 @@
+  {files, ["src/", "include/", "c_src/sqlite3_drv.c", "c_src/sqlite3_drv.h", "sqlite3_amalgamation/shell.c", "sqlite3_amalgamation/sqlite3.c", "sqlite3_amalgamation/sqlite3.h", "sqlite3_amalgamation/sqlite3ext.h", "rebar.config", "README.md", "LICENSE"]},
+  {licenses, ["EPL 1.1"]},
+  {maintainers, ["alexeyr", "ProcessOne"]},
+  {links, [{"Github", "https://github.com/processone/erlang-sqlite3"}]}]}.
```
I'd prefer this to be https://github.com/alexeyr/erlang-sqlite3/

mremond committed:
Merge pull request #2 from alexeyr/master

Merge upstream fixes

prefiks committed:
Make it compile correctly on rebar3

prefiks committed:
Include rebar.config.script in hex.pm package

prefiks committed:
Bump version

prefiks committed:
Remove erl_interface include, it's not required and causes problems on 23

prefiks committed:
Bump version to 1.1.7

badlop committed:
Update Travis setup to compile with up to 23.0. Update Rebar

mremond committed:
Merge pull request #3 from badlop/fixtravis

Update Travis setup to compile with up to 23.0. Update Rebar

badlop committed:
Bump version to 1.1.8

brigadier committed:
Handle rebar3 version strings such as "3.14.1+build.4750.refd11ceb4f"

nosnilmot committed:
Do not link to sqlite3 on macos

On macos (darwin), sqlite3 is embedded using "the amalgamation". There
is therefore no need to link to the system sqlite3, and doing so can
cause unexpected behaviour/failures.

Also be explicit about darwin vs. non-darwin port_specs sources.
rebar3 seems to get confused when compiling under eunit command sometimes

badlop committed:
Merge pull request #5 from brigadier/patch-1

Handle rebar3 version strings such as "3.14.1+build.4750.refd11ceb4f"

prefiks committed:
Release v1.1.5

prefiks committed:
Fix compilation on macos 11.0

prefiks committed:
Release v1.1.6

mremond committed:
Update version and remove deprecated attribute

prefiks committed:
Merge pull request #6 from nosnilmot/darwin-nolink

Do not link to sqlite3 on macos

prefiks committed:
Release 1.1.10

prefiks committed:
Use headers from amalagamation on darwin instead of system one

prefiks committed:
Release 1.1.11

badlop committed:
Fix wrong gen_server spec detected by Dialyzer

badlop committed:
Add Github Action to release to hex.pm when tagging

badlop committed:
Release 1.1.12

badlop committed:
Parametrize the rebar binary (processone/fast_tls#51)

badlop committed:
Switch from Travis to Github Actions

badlop committed:
Fix badge URL

badlop committed:
Use COVERALLS environment variable to detect when coveralls is required

badlop committed:
No need to set src_dirs, this way test/ isn't included in cover reports

badlop committed:
Update .gitignore

badlop committed:
Add support to send C coverage. Don't cache. Move hex plugin to Action

badlop committed:
Add configure

badlop committed:
Fix typo in gcov compilation code

prefiks committed:
Release 1.1.13

prefiks committed:
Remove rebar binary, it's better to use system one

And it even don't work with R25 because it doesn't load .beam compiled on
older version.

Fixes issue #9

prefiks committed:
Fix hex-release action

prefiks committed:
Release 1.1.14

Commit:
Added foreign_key table constraint

Commit:
Added additional string to configure update/delete actions

prefiks committed:
Merge pull request #13 from Xelynega/master

Add foreign_key table constraint

badlop committed:
CI: Test with oldest Erlang, recent stable, and the newest available

badlop committed:
Eunit: Don't wait for message

badlop committed:
CI: Test also with Erlang/OTP 27

badlop committed:
EUnit: Starting in Erlang/OTP 27.0-rc1, 0.0 and -0.0 are longer exactly equal

Reference:
https://erlang.org/documentation/doc-15.0-rc1/doc/upcoming_incompatibilities.html#0-0-and-0-0-will-no-longer-be-exactly-equal

badlop committed:
Use ubuntu-22.04, add erlang:26 container as that ubuntu doesn't include it

badlop committed:
CI: Use checkout v3 for erlang < 20.3, and v4 for newer ones

actions/checkout@v4 updated from node16 to node20,
which requires a recent GLIBC 2.27 or 2.28,
however some old erlang container images include too old GLIBC

badlop committed:
CI: For cover use setup-beam because container breaks pip

badlop committed:
CI: Install rebar3 in ubuntu-22.04 when using setup-beam

badlop committed:
CI: Use checkout v4 also in Cover job

prefiks committed:
Release 1.1.15
